app/controllers/api/v1/accounts_controller.rb

```diff
@@ -21,13 +21,17 @@ def show
   end
 
   def create
-    token    = AppSignUpService.new.call(doorkeeper_token.application, account_params)
-    response = Doorkeeper::OAuth::TokenResponse.new(token)
+    if ENV['HCAPTCHA_ENABLED'] == 'true'
+	  not_found
+	else
+      token    = AppSignUpService.new.call(doorkeeper_token.application, account_params)
+      response = Doorkeeper::OAuth::TokenResponse.new(token)
 
-    headers.merge!(response.headers)
+      headers.merge!(response.headers)
 
-    self.response_body = Oj.dump(response.body)
-    self.status        = response.status
+      self.response_body = Oj.dump(response.body)
+      self.status        = response.status
+	end
```

Comment:
The indentation is messed up, also my comment on a `before_action` hook still stands, I think that's more in line with how upstream Mastodon would do.